src/com/projectkorra/projectkorra/firebending/FireBlast.java

```diff
@@ -161,13 +162,10 @@ public boolean checkLocation(Block block) {
 				final BlastFurnace blastF = (BlastFurnace) block.getState();
 				blastF.setBurnTime((short) 800);
 				blastF.update();
-			} else if (block instanceof Campfire) {
-				final Campfire campfire = (Campfire) block.getBlockData();
-				if(!campfire.isLit()) {
-					if(block.getType() != Material.SOUL_CAMPFIRE || bPlayer.canUseSubElement(SubElement.BLUE_FIRE)) {
-						campfire.setLit(true);
-					}
-				}
+			} else if (block.getBlockData() instanceof Lightable) {
+				Lightable lightable = (Lightable) block.getBlockData();
+				lightable.setLit(true);
+				block.setBlockData(lightable);

```

Comment:
This'll change functionality right? Is that intended?

Reply:
Correct, this new simpler implementation removes the previous behavior of requiring the blue fire subelement to light soul campfires. Is this a detail we care about maintaining?

Reply:
I wasn't sure, personally I think we shouldn't keep the detail. As in minecraft blue fire is soul fire, so it would make sense that lightning a soul campfire would make the fire blue. just like using a fire charge or flint and steel lights blue fire.

Reply:
@StrangeOne101 @Dreig-Michihi what are your thoughts?

Reply:
> @StrangeOne101 @Dreig-Michihi what are your thoughts?

My thoughts about campfires:
I think you don't need a BlueFire subelement to light up soul campfires, because you can do the same with flint&steel.
I think fire blocks made with firebending (Blaze/ChargedFireBlast) should be soul fire, if they are placed on soul blocks.

If you have the blue fire sub, you should lit normal campfires as normal campfires though.
===========
About the code.
Lightable interface also used in furnaces and redstone wall torch.
![image](https://github.com/ProjectKorra/ProjectKorra/assets/60939608/ed418528-030c-4630-80cc-cbfcb406da12)
I don't think firebending should affect on redstone torches.
And I don't know, how setting lit on furnace works.
I hope it won't make the furnace visually lighted, but not mechanically?
Have to be tested in the game! =)

Also, what if mojang will add something using Lightable interface, but it won't be related to any firebending??

Reply:
Won't it also affect on such things as redtone lamp and etc?

Reply:
aren't furnaces already lit by firebending?

Reply:
> aren't furnaces already lit by firebending?

nvm